HW_2_Fastq_filtrator/fastq_filtrator.py

```diff
@@ -0,0 +1,107 @@
+def read_fastq(input_fastq):
+    names = []
+    sequences = []
+    qualities = []
+
+    with open(input_fastq, 'r') as fastq:
+        while True:
+            name = fastq.readline().strip()
+            if not name:  # End of reading the file if there is no line with the name of the read
+                break
+            sequence = fastq.readline().strip()
+            fastq.readline().strip()  # Reading without saving the technical string with the "+" sign
+            quality = fastq.readline().strip()
+
+            names.append(name)
+            sequences.append(sequence)
+            qualities.append(quality)
+
+    return names, sequences, qualities
+
+
+def write_fastq(output_file, names, sequences, qualities, indexes):
+    with open(output_file, 'w') as file:
+        for i in indexes:
+            file.write(f'{names[i]}\n')
+            file.write(f'{sequences[i]}\n')
+            file.write('+\n')
+            file.write(f'{qualities[i]}\n')
+
+
+def calculate_gc(sequence):
+    length = len(sequence)
+    if length == 0:
+        return 0
+    else:
+        g_count = sequence.upper().count("G")
+        c_count = sequence.upper().count("C")
+        return (g_count + c_count) * 100 / length
+
+
+def gc_check(sequence, gc_bounds):
+    return gc_bounds[0] <= calculate_gc(sequence) <= gc_bounds[1]
+
+
+def length_check(sequence, length_bounds):
+    return length_bounds[0] <= len(sequence) <= length_bounds[1]
+
+
+def quality_check(quality, quality_threshold, ascii_dict):
+    letters_scores = [ascii_dict[q] - 33 for q in quality]  # translate ASCII quality encoding into numeric

```

Comment:
Если бы мы использовали `ord`. СЛоварь тогда был бы не нужен
```suggestion
    letters_scores = [ord(q) - 33 for q in quality]  # translate ASCII quality encoding into numeric

```